File: SpringBoot/AnotherProject/src/main/java/com/richard/estudos/anotherproject/controllers/UsersController.java
package com.richard.estudos.anotherproject.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.richard.estudos.anotherproject.controllers.requests.UserRequest;
import com.richard.estudos.anotherproject.controllers.responses.UserResponse;
import com.richard.estudos.anotherproject.controllers.responses.UsersResponse;
import com.richard.estudos.anotherproject.models.User;
import com.richard.estudos.anotherproject.services.UsersService;

@RestController
@RequestMapping(value="/users")
public class UsersController {

	@Autowired
	private UsersService usersService;
	
	@RequestMapping(method=RequestMethod.GET, value="/")
	public ResponseEntity<UsersResponse> list() {
		Iterable<User> users = usersService.findAll();
		return new ResponseEntity<UsersResponse>(new UsersResponse(users), HttpStatus.OK);
	}
	
	@RequestMapping(method=RequestMethod.POST, value="/")
	public ResponseEntity<UserResponse> create(@RequestBody UserRequest request) {
		User userModel = request.toModel();
		User user = usersService.save(userModel);
		return new ResponseEntity<UserResponse>(new UserResponse(user), HttpStatus.CREATED);
	}
}

File: Kotlin/dynkot/src/main/kotlin/com/richard/dynkot/controllers/UsersController.kt
package com.richard.dynkot.controllers

import com.richard.dynkot.controllers.requests.CreateUserRequest
import com.richard.dynkot.controllers.responses.UserResponse
import com.richard.dynkot.entities.User
import com.richard.dynkot.services.UsersService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UsersController(private val usersService: UsersService) {

    @PostMapping
    fun create(@RequestBody createUserRequest : CreateUserRequest) : UserResponse {
        val user : User = usersService.save(createUserRequest.toModel())
        return UserResponse(user.email, user.name)
    }

    @GetMapping("/{email}")
    fun findByEmail(@PathVariable("email") email : String) : UserResponse {
        val user = usersService.findByEmail(email)
        return UserResponse(user.email, user.name)
    }
}